flash/core/data/io/input_base.py

```diff
@@ -26,6 +27,12 @@
     GenericMeta = type
 
 
+if not os.environ.get("READTHEDOCS", False):
+    from torch.utils.data import IterableDataset
+else:
+    IterableDataset = object
```

Comment:
This is the strangest thing ever. If we look at line 94:

```py
class _IterableInputMeta(_InputMeta, type(IterableDataset)):
```

which means if the if condition above holds, we will get
```py
type(IterableDataset) = torch.utils.data._typing._DataPipeMeta
```
and if it does not hold we will get
```py
type(object) = type
```

Just mentioning it in case this was not the intention.

Reply:
This is intentional :smiley: The read the docs build environment doesn't include torch so it mocks the `IterableDataset` import. Whatever type that mock is doesn't then work as a base for a metaclass. So we instead replace the import with object since `type(object) = type` and `type` is the natural base for a Python metaclass. I've added a comment to the code to make this more clear :smiley: